tests/src/JIT/Performance/CodeQuality/HWIntrinsic/X86/PacketTracer/VectorPacket.cs

```diff
@@ -0,0 +1,169 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using static System.Runtime.Intrinsics.X86.Avx;
+using static System.Runtime.Intrinsics.X86.Sse;
+using System.Runtime.Intrinsics;
+using System.Runtime.CompilerServices;
+using System;
+
+internal class VectorPacket256
+{
+    public Vector256<float> Xs;
+    public Vector256<float> Ys;
+    public Vector256<float> Zs;
+    public Vector256<float> Lengths
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get
+        {
+            var x2 = Multiply(Xs, Xs);
```

Comment:
Couldn't you simplify this to `FMA`?

Reply:
Or `DotProduct`?

Reply:
Yes, I plan to add conditional code path when I add SSE implementation.